### DISCUSSION TITLE
DOS text resolution / Mode command.

### DISCUSSION THREAD

**Discussion by bonnedav** [dosbox-staging/dosbox-staging#4034] [Q&A] [answered]
Hello,
I am using version 0.82.0 (9df43) of Dosbox-staging.
i can't figure out how to change the text resolution of DOS.
in original dosbox when i set a full screen or windowed resolution in the config the DOS text resolution changes to be close.
in staging when i set full screen resolution or resize the window the DOS resolution remains unchanged. 
I read about a mode command being added in 0.82 that looks like can change this but it just says its an illegal command for me.
How do i enable to mode command and/or automatic resizing of DOS text?
Thank you.

**Comment by MasterO2** [comment]
@bonnedav 

![image0002-rendered](https://github.com/user-attachments/assets/89e127dc-5aa1-4856-8f00-e19d8a803921)

![image0001-rendered](https://github.com/user-attachments/assets/03675fbe-9a8a-46a6-b437-4f6622afb0be)

**Reply by bonnedav** [reply]
looks like mode was not working for me because i had changed my path in autoexec without adding the default path back in. Thank you.

**Comment by johnnovak** [comment, marked as answer]
> in original dosbox when i set a full screen or windowed resolution in the config the DOS text resolution changes to be close.

It doesn't.

> in staging when i set full screen resolution or resize the window the DOS resolution remains unchanged.

You're confusing resolution with viewport size.

Try setting `integer_scaling = off` in the `render` section, then the viewport will always fill the available space in windowed/fullscreen.

But that will cause moire artifacts with the CRT shaders on 1080p and 1440p screen, so you might wanna use `glshader = sharp`.

**Reply by bonnedav** [reply]
That made resizing the window work the way i was looking for, thank you.
